### ISSUE TITLE
casting type typo

### ISSUE THREAD

**Issue by jeras** [pulp-platform/common_cells#91]
the cast should be to `idx_t` instead of `DataType`
there should be no functional changes
https://github.com/pulp-platform/common_cells/blob/master/src/rr_arb_tree.sv#L284

**Comment by zarubaf** [comment]
Yes indeed, good catch! Would you be willing to submit a PR for that? Thanks!

**Comment by jeras** [comment]
I created a pull request #94.

I have a couple of questions not really related to this pull request:

1. The RTL is written in a fully concurrent style. Usually I prefer the concurrent style in most cases, since it is easier to understand how well it will map to hardware. There are exceptions where I use a more sequential style similar to programming in C, one such example would be the [conversion between binary and Gray code](https://github.com/jeras/sockit_cdc/blob/master/rtl/sockit_cdc.v#L67-L94). I isolate the sequential code inside functions, to avoid mixing it with concurrent code.
The point is the listed concurrent code is very difficult to read, equivalent sequential code would probably be much shorter and cleaner.
Is the used coding style recommended in some Pulp platform coding style document or is this just the preferred style of the authors?

2. Is there a reference document describing this algorithm? A published research article of something less formal.

**Comment by accuminium** [comment]
> Is the used coding style recommended in some Pulp platform coding style document or is this just the preferred style of the authors?

[We have adopted](https://github.com/pulp-platform/style-guidelines) the [lowRISC Verilog Coding Style Guide](https://github.com/lowRISC/style-guides/blob/master/VerilogCodingStyle.md).
